docs/blueprints/log-in-with-storj.md

```diff
@@ -0,0 +1,186 @@
+along with the profile or "session" data so when InstaCat requests the user information in step 7, they can obtain the
+associated encryption key.
+
+<!--
+    !important challenge:
+    what if, the authorized application is a remote server? the user no longer has end to end encryption. The
+    application integrating with Storj could decrypt the stream before the data reaches it's final destination. This
+    is something we can probably consider "out of scope" as that seems beyond our responsibility.
+-->
+
+**Satellites**
+
+At this point, we have API Keys (access tokens), an encryption key and algorithm, the only remaining element is how to
+handle the satellites. This has arguably been one of the harder parts about the design. In order for client applications
+to integrate with Storj, they need to account for non-standard deployments of Storj (i.e. not just us1.storj.io, etc).
+Perhaps a good analogy here is GitLab. Sure, you can log in and develop code on gitlab.com, but it's also possible to
+run and host GitLab yourself on another custom domain. Applications integrating with GitLab need to account for this as
+they develop applications against it.
+
+Given that, we can assume that the client already has the satellite address since applications need to register with
+satellites in order to obtain a client id and client secret. As a result, applications should have all the information
+that they need in order to issue API requests.
+
+Once a client completes the OAuth workflow, they can perform requests on behalf of the end user for as long as the token
+is valid. To Alice, they're just interacting with this new InstaCat application, but they own all their data.
+
+![common request flow](images/log-in-with-storj-request.svg)
+
+## Rationale
+
+_Why OpenID Connect?_
+
+OpenID Connect has become a fairly strong (and consistent) standard for handling user authentication. Google Accounts,
+Salesforce, and Azure AD v2 are prominent examples out in the wild today. While there are many other types of mechanisms
+out there, this one immediately came to mind because of its ubiquity across major platforms. For solutions that don't
+implement OIDC, [dex][] has become a popular tool of choice that can convert many types of authentication flows into a
+provider. I've also seen [hydra][], which seems to allow companies to bolt and OIDC provider onto the side of an
+existing login workflow.
+
+In the end, making the decision to go with OIDC seemed like the right thing to do given its rather extensive use and our
+longer term vision of a more decentralized web.
+
+[dex]: https://github.com/dexidp/dex/
+
+[hydra]: https://github.com/ory/hydra
+
+## Implementation
+
+- Satellite Changes
+    - API / OIDCP
+        - New OAuth2 and OpenID endpoints to properly support authenticating users
+        - Database or Redis for token/code storage
```

Comment:
@egonelbre: what's our preferred AP datastore? Using something like postgres/cockroach seems out of place for _relatively_ short-lived tokens.

Reply:
We will want something more durable for storing client applications, but this could honestly be a hard-coded list for now.

For tokens, I've been leaning toward redis. It's relatively easy for folks to run and configure on their own. We can configure it a few different ways depending on the requirements of the cluster.